From 8e5de45513577c63a33833931f2afd0cc59d42a7 Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Date: Wed, 24 Feb 2010 02:56:59 -0600
Subject: [PATCH 10/37] Use libtool -export-dynamic flag for portability

The linker flag required for exporting symbols in executables varies
by platform.  libtool handles this with a single -export-dynamic
flag (not to be confused with the similarly-named ELF linker flag)
which tells it to use the correct platform-specific flag at link time.

Signed-off-by: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Reviewed-by: Dan Nicholson <dbn.lists@gmail.com>
Reviewed-by: Keith Packard <keithp@keithp.com>
---
 configure.ac |   14 +++-----------
 1 files changed, 3 insertions(+), 11 deletions(-)

diff --git xorg-server-1.7.99.902/configure.ac xorg-server-1.7.99.902/configure.ac
index e497e76..d379b3a 100644
--- xorg-server-1.7.99.902/configure.ac
+++ xorg-server-1.7.99.902/configure.ac
@@ -1440,17 +1440,9 @@ UTILS_SYS_LIBS="${SYS_LIBS}"
 AC_SUBST([UTILS_SYS_LIBS])
 
 # The Xorg binary needs to export symbols so that they can be used from modules
-# Some platforms require extra flags to do this.   gcc should set these flags
-# when -rdynamic is passed to it, other compilers/linkers may need to be added
-# here.
-if test "x$GCC" = "xyes"; then
-       LD_EXPORT_SYMBOLS_FLAG="-rdynamic"
-fi
-case $host_os in
-       openbsd*)
-       LD_EXPORT_SYMBOLS_FLAG="-Wl,--export-dynamic"
-       ;;
-esac
+# Some platforms require extra flags to do this.   libtool should set the
+# necessary flags for each platform when -export-dynamic is passed to it.
+LD_EXPORT_SYMBOLS_FLAG="-export-dynamic"
 AC_SUBST([LD_EXPORT_SYMBOLS_FLAG])
 
 dnl Imake defines SVR4 on SVR4 systems, and many files check for it, so
-- 
1.7.0.2

